services/galley/src/Galley/API/Teams/Features.hs

```diff
@@ -546,31 +569,53 @@ getConferenceCallingInternal (Right tid) = do
 setConferenceCallingInternal ::
   Members '[GundeckAccess, TeamFeatureStore, TeamStore, P.TinyLog] r =>
   TeamId ->
-  Public.TeamFeatureStatus 'Public.TeamFeatureConferenceCalling ->
-  Sem r (Public.TeamFeatureStatus 'Public.TeamFeatureConferenceCalling)
+  Public.TeamFeatureStatus 'Public.WithoutPaymentStatus 'Public.TeamFeatureConferenceCalling ->
+  Sem r (Public.TeamFeatureStatus 'Public.WithoutPaymentStatus 'Public.TeamFeatureConferenceCalling)
 setConferenceCallingInternal =
   setFeatureStatusNoConfig @'Public.TeamFeatureConferenceCalling $ \_status _tid -> pure ()
 
 getSelfDeletingMessagesInternal ::
   Member TeamFeatureStore r =>
   GetFeatureInternalParam ->
-  Sem r (Public.TeamFeatureStatus 'Public.TeamFeatureSelfDeletingMessages)
+  Sem r (Public.TeamFeatureStatus 'Public.WithPaymentStatus 'Public.TeamFeatureSelfDeletingMessages)
 getSelfDeletingMessagesInternal = \case
   Left _ -> pure Public.defaultSelfDeletingMessagesStatus
-  Right tid ->
-    TeamFeatures.getSelfDeletingMessagesStatus tid
-      <&> maybe Public.defaultSelfDeletingMessagesStatus id
+  Right tid -> do
+    (maybeFeatureStatus, maybePaymentStatus) <- TeamFeatures.getSelfDeletingMessagesStatus tid
+    pure $ case (maybePaymentStatus, maybeFeatureStatus) of
+      (Just Public.PaymentUnlocked, Just featureStatus) ->
+        Public.TeamFeatureStatusWithConfigAndPaymentStatus
+          (Public.tfwcStatus featureStatus)
+          (Public.tfwcConfig featureStatus)
+          Public.PaymentUnlocked
+      (Just Public.PaymentUnlocked, Nothing) ->
+        Public.TeamFeatureStatusWithConfigAndPaymentStatus
+          (Public.tfwcapsStatus Public.defaultSelfDeletingMessagesStatus)
+          (Public.tfwcapsConfig Public.defaultSelfDeletingMessagesStatus)
+          Public.PaymentUnlocked
+      _ -> Public.defaultSelfDeletingMessagesStatus
```

Comment:
```suggestion
      (Just Public.PaymentUnlocked, _) -> Public.defaultSelfDeletingMessagesStatus
```

Does the same thing, but it explains better and gives warnings if we extend the types and add cases, and forget to handle them here.